cob/cob.c

```diff
@@ -877,6 +1070,69 @@ static int cob_ns_lookup(struct m0_cob *cob);
 static int cob_oi_lookup(struct m0_cob *cob);
 static int cob_fab_lookup(struct m0_cob *cob);
 
+M0_INTERNAL int m0_cob_bc_lookup(struct m0_cob *cob,
+				 struct m0_cob_bckey *bc_key)
+{
+	struct m0_buf key;
+	struct m0_buf val;
+	int           rc;
+
+	M0_PRE(bc_key != NULL &&
+	       m0_fid_is_set(&bc_key->cbk_pfid));
+
+	m0_buf_init(&key, bc_key, m0_cob_bckey_size(bc_key));
+	m0_buf_init(&val, &cob->co_bcrec, sizeof cob->co_bcrec);
```

Comment:
Instead of populating cob->co_bcrec, lets add an out parameter for rec and populate that during lookup.